File: 3 Loops/4Nested_Loop4_Pyramid.js
function printPattern(n) {
    for (let i = 0; i < n; i++) { // Outer loop
        // Spaces n - i - 1
        let line = ' '.repeat(n - i - 1); // Create spaces at the start of the line

        // Numbers 1: i + 1
        for (let j = 1; j < i + i + 1; j++) {
            line += j; // Add numbers incrementally
        }

        // Numbers 2: reverse
        for (let j = i; j >= 0; j--) {
            line += j; // Add numbers in reverse
        }

        console.log(line); // Print the final line for this row
    }
}

printPattern(4);

File: 3 Loops/4Nested_Loop4_Pyramid.py
def print_pattern(n):
    for i in range(n):  # Outer loop
        # Create spaces n - i - 1
        line = ' ' * (n - i - 1)  # Generate the leading spaces

        # Numbers 1: from 1 to i + 1
        for j in range(1, i + 2):  # Loop to print numbers incrementally from 1 to i+1
            line += str(j)

        # Numbers 2: reverse from i to 1
        for j in range(i, 0, -1):  # Loop to print numbers in reverse from i to 1
            line += str(j)

        # Print the final line for this row
        print(line)

# Example usage
print_pattern(4)
